### PR TITLE
Fix (most) warnings

### PR DESCRIPTION
**Pull request by ChiragCD**
# Why?

The "mix compile --force" command generates a number of warnings. Many of them are in dependencies, and cannot be resolved here. This merge handles most of the remaining warnings.

## Significant Changes

- Add local (umbrella) and general dependencies to mix files
- Set core/schema functions (add_uuid, add_slug, random_string) as overridable, as specific implementations are provided for these in some files. These implementations were previously ignored, they will now be used. May cause changes in dashboard management and tool management, may require testing.
- In pivot_tables.ex, private function pivot_values_col_data, set acc to accumulate all rows, rather than just the last row from rows_data.

## Minor Changes

- Convert variables to anonymous variables where they are not used
- Comment out import/use/alias statements where they are not used
- Reorder functions to place those with same name and arity together

### CODE REVIEW

**Review by blackode — CHANGES_REQUESTED** [review]

File: apps/acqdat_api/lib/acqdat_api/alerts/alert_filter_listing.ex [hunk 1]
```diff
@@ -3,13 +3,13 @@ defmodule AcqdatApi.Alerts.AlertFilterListing do
...
+    Enum.reduce(AlertStatusEnum.__enum_map__(), [], fn {key, _value}, acc ->
       acc ++ [key]
     end)
   end
 
   def list_app() do
-    Enum.reduce(AppEnum.__enum_map__(), [], fn {key, value}, acc ->
+    Enum.reduce(AppEnum.__enum_map__(), [], fn {key, _value}, acc ->
       acc ++ [key]
     end)
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_api/lib/acqdat_api/alerts/alert_filter_listing.ex]: Implemented

File: apps/acqdat_api/lib/acqdat_api/alerts/alert_filter_listing.ex [hunk 2]
```diff
@@ -3,13 +3,13 @@ defmodule AcqdatApi.Alerts.AlertFilterListing do
   All the helper functions related to alert rules listing.
   """
   def list_status() do
-    Enum.reduce(AlertStatusEnum.__enum_map__(), [], fn {key, value}, acc ->
+    Enum.reduce(AlertStatusEnum.__enum_map__(), [], fn {key, _value}, acc ->
       acc ++ [key]
     end)
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_api/lib/acqdat_api/alerts/alert_filter_listing.ex]: Implemented

File: apps/acqdat_api/lib/acqdat_api/alerts/policy.ex [hunk 3]
```diff
@@ -4,7 +4,7 @@ defmodule AcqdatApi.Alerts.Policy do
   """
 
   def list_policies() do
-    Enum.reduce(PolicyDefinitionModuleEnum.__enum_map__(), [], fn {key, value}, acc ->
+    Enum.reduce(PolicyDefinitionModuleEnum.__enum_map__(), [], fn {key, _value}, acc ->
       acc ++ [key]
     end)
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_api/lib/acqdat_api/alerts/policy.ex]: Implemented

File: apps/acqdat_api/lib/acqdat_api/api_access/user_group.ex [hunk 4]
```diff
@@ -30,6 +30,11 @@ defmodule AcqdatApi.ApiAccess.UserGroup do
     add_policies(total_policy_ids, params, group)
   end
 
+  def update(group, params) do
+    params = for {key, val} <- params, into: %{}, do: {String.to_atom(key), val}
+    verify_group(UserGroup.normal_update(group, params))
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_api/lib/acqdat_api/api_access/user_group.ex]: Implemented

File: apps/acqdat_api/lib/acqdat_api/data_cruncher/task.ex [hunk 5]
```diff
@@ -33,11 +27,17 @@ defmodule AcqdatApi.DataCruncher.Task do
     |> run_transaction()
   end
 
+  def update(%{"id" => id, "action" => action} = params)
+      when action == "register" do
+    {id, _} = Integer.parse(id)
+    verify_task(TaskModel.get(id), params)
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_api/lib/acqdat_api/data_cruncher/task.ex:33]: Possible consistency issues?

File: apps/acqdat_api/lib/acqdat_api/data_insights/fact_tables.ex [hunk 6]
```diff
@@ -211,7 +211,7 @@ defmodule AcqdatApi.DataInsights.FactTables do
           [[key] ++ List.flatten(Enum.map(metadatas, fn metadata -> Map.values(metadata) end))]
       end)
 
-    output =
+    _output =
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_api/lib/acqdat_api/data_insights/fact_tables.ex]: Implemented

File: apps/acqdat_api/lib/acqdat_api/data_insights/fact_tables.ex [hunk 7]
```diff
@@ -298,7 +298,7 @@ defmodule AcqdatApi.DataInsights.FactTables do
         {headers, data1}
       end
 
-    output =
+    _output =
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_api/lib/acqdat_api/data_insights/fact_tables.ex]: Implemented

File: apps/acqdat_api/lib/acqdat_api/data_insights/fact_tables.ex [hunk 8]
```diff
@@ -381,7 +381,7 @@ defmodule AcqdatApi.DataInsights.FactTables do
         }
       end
 
-    output =
+    _output =
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_api/lib/acqdat_api/data_insights/fact_tables.ex]: Implemented

File: apps/acqdat_api/lib/acqdat_api_web/controllers/entity-management/project_controller.ex [hunk 9]
```diff
@@ -246,7 +246,7 @@ defmodule AcqdatApiWeb.EntityManagement.ProjectController do
   end
 
   defp modify_params(conn, params) do
-    params =
+    _params =
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/entity-management/project_controller.ex]: Implemented

File: apps/acqdat_core/lib/acqdat_core/data_cruncher/domain/task.ex [hunk 10]
```diff
@@ -29,7 +29,7 @@ defmodule AcqdatCore.DataCruncher.Domain.Task do
 
   ############################# private functions ###########################
 
-  defp create_graph(%{graph: graph} = workflow) do
+  defp create_graph(%{graph: graph} = _workflow) do
     edge_list =
       Enum.reduce(graph["edge_list"], [], fn edge, acc ->
         acc ++ [gen_edge(edge)]
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_core/lib/acqdat_core/data_cruncher/domain/task.ex]: Implemented

File: apps/acqdat_core/priv/repo/seed/data_insights/topology.ex [hunk 11]
```diff
@@ -353,6 +353,38 @@ defmodule AcqdatCore.Seed.DataInsights.Topology do
...
 
+  def gen_sensor_type_data("occupancy", sensor) do
+    duration = @months * 30 # duration in months
+    interval = 10 # interval in minutes
+    iterator = get_time_iterator(duration, interval)
+
+    Enum.map(iterator, fn time ->
+      time = time |> DateTime.from_naive!("Etc/UTC") |> DateTime.truncate(:second)
+      %{
+        sensor_id: sensor.id, project_id: sensor.project_id, org_id: sensor.org_id,
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_core/priv/repo/seed/data_insights/topology.ex]: Implemented

File: apps/acqdat_core/priv/repo/seed/data_insights/topology.ex [hunk 12]
```diff
@@ -353,6 +353,38 @@ defmodule AcqdatCore.Seed.DataInsights.Topology do
...
+
+  def gen_sensor_type_data("heat", sensor) do
+    duration = @months * 30 # duration in months
+    interval = 10 # interval in minutes
+    iterator = get_time_iterator(duration, interval)
+
+    Enum.map(iterator, fn time ->
+      time = time |> DateTime.from_naive!("Etc/UTC") |> DateTime.truncate(:second)
+      %{
+        sensor_id: sensor.id, project_id: sensor.project_id, org_id: sensor.org_id,
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_core/priv/repo/seed/data_insights/topology.ex]: Implemented

File: apps/acqdat_core/priv/repo/seed/data_insights/topology.ex [hunk 13]
```diff
@@ -374,49 +406,17 @@ defmodule AcqdatCore.Seed.DataInsights.Topology do
         result = create_parameter_struct(params, energy)
         Map.put(acc, "Energy", result)
     end)
-    |> Enum.map(fn {key, value} ->
+    |> Enum.map(fn {_key, value} ->
       value
     end)
```
[the remark this answers is not in the document]
Comment (ChiragCD) [reply apps/acqdat_core/priv/repo/seed/data_insights/topology.ex]: Implemented

**Comment by ChiragCD** [comment]
Subsumed under PR 220, branch fixes/warnings

**Pull request closed without merging**